File: 0978_longest_turbulent_subarray/solution.py
# LeetCode 0978 - Longest Turbulent Subarray
# https://leetcode.com/problems/longest-turbulent-subarray/

class Solution:
    def maxTurbulenceSize(self, arr: list[int]) -> int:
        ans = cur = 1
        for i in range(1, len(arr)):
            if arr[i] == arr[i - 1]:
                cur = 1
            elif i == 1 or (arr[i] - arr[i - 1]) * (arr[i - 1] - arr[i - 2]) < 0:
                cur += 1
            else:
                cur = 2
            ans = max(ans, cur)
        return ans

File: 0978_longest_turbulent_subarray/solution.rs
// LeetCode 0978 - Longest Turbulent Subarray
// https://leetcode.com/problems/longest-turbulent-subarray/

impl Solution {
    pub fn max_turbulence_size(arr: Vec<i32>) -> i32 {
        let mut ans = 1;
        let mut cur = 1;
        for i in 1..arr.len() {
            if arr[i] == arr[i - 1] {
                cur = 1;
            } else if i == 1
                || (arr[i] - arr[i - 1]) as i64 * (arr[i - 1] - arr[i - 2]) as i64 < 0
            {
                cur += 1;
            } else {
                cur = 2;
            }
            ans = ans.max(cur);
        }
        ans
    }
}
